coalib/core/PersistentHash.py

```diff
@@ -1,7 +1,53 @@
+from collections import Iterable
 from hashlib import sha1
 import pickle
 
 
+def tuple_replace(tup, x, y):
+    tup_list = list(tup)
+    for element in tup_list:
+        if element == x:
+            tup_list[tup_list.index(element)] = y
```

Comment:
Use `list.index` instead to find elements. That's probably way faster since it should utilize C code.